packages/block-editor/src/store/reducer.js

```diff
@@ -1800,6 +1800,15 @@ export function editorMode( state = 'edit', action ) {
 		return 'edit';
 	}
 
+	// Let inserting a single block always trigger Edit mode.
+	if (
+		action.type === 'INSERT_BLOCKS' &&
+		action.blocks.length === 1 &&
+		action.blocks[ 0 ].innerBlocks.length === 0
+	) {
+		return 'edit';
+	}
+
```

Comment:
I'm noting that in the line above the behaviour is conditionalised to `navigation` mode.

https://github.com/WordPress/gutenberg/blob/54269c00f74805533b5fec9bf14b84d18f8d3ae9/packages/block-editor/src/store/reducer.js#L1799-L1803

This action will apply to all modes in all editors. Are we confident that's safe? Would it be better to limit the action to be in Zoom Out mode only?
```suggestion
	// Let inserting a single block always trigger Edit mode.
	if (
		state === 'zoom-out' && // only applies in zoom out mode
		action.type === 'INSERT_BLOCKS' &&
		action.blocks.length === 1 &&
		action.blocks[ 0 ].innerBlocks.length === 0
	) {
		return 'edit';
	}

```